kedro/io/partitioned_data_set.py

```diff
@@ -53,6 +54,13 @@
 S3_PROTOCOLS = ("s3", "s3a", "s3n")
 
 
+def _grandparent(path: str) -> str:
+    path_obj = Path(path)
+    grandparent = path_obj.parent.parent
+    assert grandparent.name == path_obj.name
```

Comment:
`assert`s don't provide a particularly informative error message. Can you rather raise a `DataSetError`?

Reply:
Done!